app/scripts/router.js

```diff
@@ -139,7 +141,7 @@ function (
       // back is enabled after the first view is rendered or
       // if the user is re-starts the app.
       this.canGoBack = this.window.sessionStorage.canGoBack || false;
-
+      this.environment = new Environment(this.window);
```

Comment:
Huh, the other private variables do not have the `_` prefix. Hmmm..  That should be changed, but not in this PR.